SimPEG/electromagnetics/static/utils/static_utils.py

```diff
@@ -258,291 +297,260 @@ def geometric_factor(survey_object, space_type="half space", **kwargs):
+        if output_indexing:
+            out_indices_list.append(np.hstack(out_indices))
+    
+    if output_indexing:
+        return survey_list, out_indices_list
     else:
-        midx = midx[:, 0]
+        return survey_list
 
-    if data_type.lower() in (
-        DATA_TYPES["potential"]
-        + DATA_TYPES["apparent chargeability"]
-        + ["misfit", "misfitmap"]
-    ):
-        if scale == "linear":
-            rho = dobs
-        elif scale == "log":
-            rho = np.log10(abs(dobs))
 
-    elif data_type.lower() in DATA_TYPES["apparent conductivity"]:
-        rhoApp = apparent_resistivity(data, dobs=dobs, space_type=space_type)
-        if scale == "linear":
-            rho = 1.0 / rhoApp
-        elif scale == "log":
-            rho = np.log10(1.0 / rhoApp)
+#####################################################################
+#                               PLOTTING
+#####################################################################
 
-    elif data_type.lower() in DATA_TYPES["apparent resistivity"]:
-        rhoApp = apparent_resistivity(data, dobs=dobs, space_type=space_type)
-        if scale == "linear":
-            rho = rhoApp
-        elif scale == "log":
-            rho = np.log10(rhoApp)
 
-    else:
-        print()
-        raise Exception(
-            """data_type must be 'potential' | 'apparent resistivity' |
-                'apparent conductivity' | 'apparent chargeability' | misfit"""
-            " not {}".format(data_type)
-        )
+def plot_1d_layer_model(
+    thicknesses, values, z0=0, scale="log", ax=None, plot_elevation=False, show_layers=False, **kwargs
+):
+    """
+    Plot the vertical conductivity or resistivity profile for a 1D layered Earth model.
+    
+    Input:
+    thicknesses : List[Float]
+        A list or numpy.array containing the layer thicknesses from the top layer down
+    values : List[Float]
+        A list or numpy.array containing the physical property values from the top layer down
+    z0 : Float
+        Elevation of the surface
+    scale: str
+        scale {'linear', 'log'}. Plot physical property values on a linear or log10 scale.
+    ax: mpl_toolkits.mplot3d.axes3d.Axes3D, optional
+        A 3D axis object for the 3D plot
+    plot_elevation : bool
+        If False, the yaxis will be the depth. If True, the yaxis is the elevation.
+    show_layers : bool
+        Plot horizontal lines to denote layers.
+    line_opts : dict
+        Dictionary defining kwargs for scatter plot if plot_type='scatter'
+    
 
-    # Grid points
-    grid_x, grid_z = np.mgrid[np.min(midx) : np.max(midx), np.min(midz) : np.max(midz)]
+    Output:
+    mpl_toolkits.mplot3d.axes3d.Axes3D
+        The axis object that holds the plot
 
-    grid_rho = griddata(np.c_[midx, midz], rho.T, (grid_x, grid_z), method="linear")
+    """
 
-    if clim is None:
-        vmin, vmax = rho.min(), rho.max()
+    if np.median(values) > 1.:
```

Comment:
just trying to guess at what the property is? I like that this is a general function for plotting 1D models. I would like to see it moved to some other utility file, and then just imported here for backwards compatibility.

Reply:
I moved this to general plotting functions.